src/styles/_catalog.scss

```diff
@@ -1,22 +1,39 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: repeat(4, 200px);
+  gap: 48px;
+  justify-content: center;
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media (min-width: 1024px) {
+    & {
+      grid-template-columns: repeat(4, 200px);
+    }
+  }
+
+  @media (min-width: 768px) and (max-width: 1023px) {
+    & {
+      grid-template-columns: repeat(3, 200px);
+    }
+  }
+
+  @media (min-width: 488px) and (max-width: 767px) {
+    & {
+      grid-template-columns: repeat(2, 200px);
+    }
+  }
+
+  @media (max-width: 487px) {
+    & {
+      grid-template-columns: repeat(1, 200px);
+    }
+  }
```

Comment:
Better start from mobile mode and move to desktop mode with  `min-width`.
Example 
- (min-width: 488px) - 2 cards
- (min-width: 768px) - 3 cards

It helps you avoid redundant code with `max-width`